chainercv/datasets/image_folder_dataset.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import os
+
+import chainer
+from chainercv.utils import read_image
+
+
+def find_label_names(root):
+    label_names = [d for d in os.listdir(root)
+                   if os.path.isdir(os.path.join(root, d))]
+    label_names.sort()
+    return label_names
+
+
+def _ends_with_img_ext(filename):
+    img_extensions = [
+        '.jpg', '.JPG', '.jpeg', '.JPEG', '.png', '.PNG',
+        '.ppm', '.PPM', '.bmp', '.BMP',
+    ]
+    return any(filename.endswith(extension) for extension in img_extensions)
+
+
+def _parse_classification_dataset(root, label_names,
+                                  check_img_file=_ends_with_img_ext):
+    # Use label_name_to_idx for performance.
+    label_name_to_idx = {label_names[i]: i for i in range(len(label_names))}
+
+    img_paths = []
+    labels = []
+    for label_name in os.listdir(root):
+        label_dir = os.path.join(root, label_name)
+        if not os.path.isdir(label_dir):
+            continue
+
+        for cur_dir, _, filenames in sorted(os.walk(label_dir)):
+            for filename in filenames:
+                if check_img_file(filename):
+                    img_paths.append(os.path.join(cur_dir, filename))
+                    labels.append(label_name_to_idx[label_name])
+
+    return img_paths, np.array(labels, np.int32)
+
+
+class ImageFolderDataset(chainer.dataset.DatasetMixin):
```

Comment:
`ClassificationDatasetFromDirectory` is OK, but this breaks the rule that dataset object class ends with "Dataset".

Perhaps, `FolderParsingClassificationDataset`, `DirectoryParsingClassificationDataset` or `ParsingClassificationDataset`.